src/test/kotlin/io/snyk/plugin/services/download/CliDownloaderErrorHandlerTest.kt

```diff
@@ -4,79 +4,57 @@ import com.intellij.notification.NotificationAction
 import com.intellij.openapi.progress.ProgressIndicator
 import com.intellij.openapi.project.Project
 import com.intellij.util.io.HttpRequests
+import io.mockk.CapturingSlot
 import io.mockk.clearAllMocks
 import io.mockk.every
 import io.mockk.mockk
 import io.mockk.mockkObject
 import io.mockk.slot
+import io.mockk.unmockkAll
 import io.mockk.verify
-import io.snyk.plugin.getCliFile
 import io.snyk.plugin.ui.SnykBalloonNotificationHelper
 import junit.framework.TestCase.assertEquals
 import org.junit.After
 import org.junit.Before
 import org.junit.Test
-import java.io.IOException
 
 class CliDownloaderErrorHandlerTest {
 
+    private lateinit var indicator: ProgressIndicator
+    private lateinit var project: Project
     private lateinit var cut: CliDownloaderErrorHandler
 
+    private lateinit var retryActionSlot: CapturingSlot<NotificationAction>
+    private lateinit var contactActionSlot: CapturingSlot<NotificationAction>
+    private lateinit var messageSlot: CapturingSlot<String>
+    private lateinit var projectSlot: CapturingSlot<Project>
+
     @Before
     fun setUp() {
-        cut = CliDownloaderErrorHandler()
+        clearAllMocks()
+        unmockkAll()
```

Comment:
Is there a need to call `unmockkAll` in both before and after function?

Reply:
Yes, because after the last test we would like to unmock, too, and we want to start with a clean slate for each test.

Reply:
Isn't it what tearDown function already does for you? 👀

Reply:
In general that's true - but not if you hard-stop the testing. In this case, if no VM is kept running it would probably be fine. I've seen it with database and filesystem changes though, that you're better off cleaning up before and after a test, to minimize potential for flakiness.

Reply:
Fair point, thanks for explaining. Resolving as it doesn't harm anyway :)